src/Miki/Modules/CustomCommands/CommandHandlers/CustomCommandsHandler.cs

```diff
@@ -74,48 +23,53 @@ public async ValueTask CheckAsync(IDiscordMessage data, IMutableContext e, Func<
 
             if (e.GetMessage().Type != DiscordMessageType.DEFAULT)
             {
+                await next();
                 return;
             }
 
-            var channel = e.GetChannel();
-            if (!(channel is IDiscordGuildChannel guildChannel))
-            {
-                return;
-            }
-
-            var guild = await guildChannel.GetGuildAsync();
-
-            var cache = e.GetService<IExtendedCacheClient>();
-            IEnumerable<Token> tokens = null;
-
-            string[] args = e.GetMessage().Content.Substring(e.GetPrefixMatch().Length).Split(' ');
-            string commandName = args.FirstOrDefault().ToLowerInvariant();
-
-            var cachePackage = await cache.HashGetAsync<ScriptPackage>(
-                CommandCacheKey, commandName + ":" + guild.Id);
+            var service = e.GetService<ICustomCommandsService>();
+            var startIndex = e.GetPrefixMatch().Length;
+            var message = e.GetMessage().Content;
+            var endIndex = message.IndexOf(' ', startIndex);
+            var commandName = endIndex == -1
+                ? message.Substring(startIndex)
+                : message.Substring(startIndex, endIndex - startIndex);
 
-            if (cachePackage != null)
+            try
             {
-                tokens = ScriptPacker.Unpack(cachePackage);
-            }
-            else
-            {
-                var db = e.GetService<IUnitOfWork>();
-                var repository = db.GetRepository<CustomCommand>();
+                var success = await service.ExecuteAsync(e, commandName);
 
-                var command = await repository.GetAsync((long)guild.Id, commandName);
-                if (command != null)
+                if (!success)
                 {
-                    tokens = new Tokenizer().Tokenize(command.CommandBody);
+                    await next();
```

Comment:
If not successful shouldn't it throw an exception?

Reply:
The return value is if the command was executed, if a custom command doesn't exists it returns false. It throws an exception whenever something went wrong and that's being handled a few lines down.